src/main/java/uk/ac/ebi/ega/egafuse/service/EgaFuse.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.service;
+
+import java.nio.file.Paths;
+
+import jnr.ffi.Pointer;
+import jnr.ffi.types.off_t;
+import jnr.ffi.types.size_t;
+import ru.serce.jnrfuse.ErrorCodes;
+import ru.serce.jnrfuse.FuseFillDir;
+import ru.serce.jnrfuse.FuseStubFS;
+import ru.serce.jnrfuse.struct.FileStat;
+import ru.serce.jnrfuse.struct.FuseFileInfo;
+
+public class EgaFuse extends FuseStubFS {
```

Comment:
As I've said in another commentary. This class should not require to use an empty directory to behave correctly. Please fix as that is just a hack.

Reply:
Done